spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala

```diff
@@ -177,11 +189,6 @@ class IcebergSparkSqlExtensionsParser(delegate: ParserInterface) extends ParserI
 
   private def isIcebergCommand(sqlText: String): Boolean = {
     val normalized = sqlText.toLowerCase(Locale.ROOT).trim()
```

Comment:
Under the new suggested form of first normalizing the sql text in a separate function and then passing it to this function, I chose to leave `toLowerCase` inside of this function because having the input not be lowercased would potentially break the check.

As mentioned, I prefer to keep the logic all within `isIcebergCommand`, given that we don't want to use the normalized text outside of this function / this specific purpose.

That said, if you both agree that the named function is better (despite possibly being unsafe), I can move all of the normalization into the new function and leave just the boolean check in `isIcebergCommand`.

Reply:
Resolving this as we've decided that having the `normalizeSqlText` function exist outside the scope of `isIcebergCommand` could potentially lead to issues as the antlr parser should be responsible for actually processing the sqlText.